### PR TITLE
Reth > IndexDB experiment

### PR DESCRIPTION
**Pull request by dcposch**
## Summary

See if we can feed IndexDB directly via an exex.

## Progress

Basic indexing works. To ship to prod, we'd need:
- [ ] Rest of IndexDB, not just `names`
- [ ] Support for RPC fallback

### Quick start

You need four parts: L2 EL (reth+exex), L2 CL (op-node), L1 EL (= can use standard hosted JSON RPC instead of a node), L1 CL (= can use Publicnode hosted beacon chain API).

**op-node**

Get op-node docker. Create `/root/[jwt.hex,start.sh]`. start.sh:

```bash
export RETH_HOST=... # local dev: host ip addr outside docker
export ALCHEMY_KEY=...

/usr/local/bin/op-node \
 --network="base-sepolia" \
 --l1=https://eth-sepolia.g.alchemy.com/v2/$ALCHEMY_KEY \
 --l2=http://$RETH_HOST:9551 \
 --l2.jwt-secret=/root/jwt.hex \
 --rpc.addr=0.0.0.0 \
 --rpc.port=7000 \
 --l1.beacon=https://ethereum-sepolia-beacon-api.publicnode.com \
 --syncmode=execution-layer \
 --l2.enginekind=reth \
 --l1.trustrpc
```

**reth+exex**

check out this branch. then:

```bash
cargo run --bin daimo-names-exex -- node --chain base-sepolia  --http --ws --authrpc.addr 0.0.0.0 --authrpc.port 9551 --authrpc.jwtsecret ./jwt.hex --rollup.sequencer-http https://sepolia-sequencer.base.org/  -vvvv --enable-discv5-discovery
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -19,20 +20,22 @@ publish = false
...
 reth-exex = { git = "https://github.com/paradigmxyz/reth", features = ["serde"], version = "1.0.4" }
 reth-network-peers = { git = "https://github.com/paradigmxyz/reth", version = "1.0.4" }
 reth-node-api = { git = "https://github.com/paradigmxyz/reth", version = "1.0.4" }
 reth-node-ethereum = { git = "https://github.com/paradigmxyz/reth", version = "1.0.4" }
-reth-primitives = { git = "https://github.com/paradigmxyz/reth", version = "1.0.4" }
-reth-provider = { git = "https://github.com/paradigmxyz/reth", version = "1.0.4" }
-reth-revm = { git = "https://github.com/paradigmxyz/reth", version = "1.0.4" }
+reth-primitives = { git = "https://github.com/paradigmxyz/reth", features = ["optimism"], version = "1.0.4" }
+reth-provider = { git = "https://github.com/paradigmxyz/reth", features = ["optimism"], version = "1.0.4" }
+reth-revm = { git = "https://github.com/paradigmxyz/reth", features = ["optimism"], version = "1.0.4" }
```
Comment (Rjected) [inline Cargo.toml:34]: the `features = ["optimism"]` in this file should be enabled in the `daimo-names/Cargo.toml` instead

File: daimo_names.db
Comment (Rjected) [inline daimo_names.db:1]: assuming this doesn't need to be committed, should rm or include a `daimo-names/.gitignore`

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 # will have compiled files and executables
 debug/
 target/
+jwt.hex
```
Comment (Rjected) [inline .gitignore:5]:
```suggestion
```

**Review by Rjected — CHANGES_REQUESTED** [review]
nice! some initial comments about deps / features

**Comment by dcposch** [comment]
@Rjected my b, didn't mean to send this as an upstream PR.

**Pull request closed without merging**